src/adapters/model_mixin.py

```diff
@@ -1229,23 +1322,120 @@ def average_adapter(
                 sum_weights = 1.0
             input_adapters = {name: weight / sum_weights for name, weight in zip(adapter_list, weights)}
         try:
-            self.apply_to_adapter_layers(lambda i, layer: layer.average_adapter(adapter_name, input_adapters))
-            self.apply_to_basemodel_childs(lambda i, child: child.average_adapter(adapter_name, input_adapters))
+            self.apply_to_adapter_layers(
+                lambda i, layer: layer.average_adapter(
+                    adapter_name, input_adapters, combine_strategy, svd_rank=svd_rank
+                )
+            )
+            self.apply_to_basemodel_childs(
+                lambda i, child: child.average_adapter(
+                    adapter_name, input_adapters, combine_strategy, svd_rank=svd_rank
+                )
+            )
             # PHM Layer
             if self.adapters_config.match(adapter_name, BnConfig, location_key="phm_layer"):
-                self._average_shared_parameters(adapter_name, input_adapters)
+                self._average_shared_parameters(adapter_name, input_adapters, combine_strategy)
             # Prefix Tuning
             for module in self.modules():
                 if isinstance(module, PrefixTuningPool):
-                    module.average_prefix(adapter_name, input_adapters)
+                    module.average_prefix(adapter_name, input_adapters, combine_strategy)
             if isinstance(self, InvertibleAdaptersMixin) or isinstance(self, InvertibleAdaptersWrapperMixin):
-                self._average_invertible_adapter(adapter_name, input_adapters)
+                self._average_invertible_adapter(adapter_name, input_adapters, combine_strategy)
         except ValueError as ex:
             self.delete_adapter(adapter_name)
             raise ex
         if set_active:
             self.set_active_adapters(adapter_name)
 
+    def average_head(
```

Comment:
Should this method live in the flexible heads mixin since it only applies to those model classes?